src/main/java/WordFrequencyGame.java

```diff
@@ -1,76 +1,59 @@
-                for (String s : arr) {
-                    Input input = new Input(s, 1);
-                    inputList.add(input);
-                }
-
-                //get the map for the next step of sizing the same word
-                Map<String, List<Input>> map =getListMap(inputList);
-
-                List<Input> list = new ArrayList<>();
-                for (Map.Entry<String, List<Input>> entry : map.entrySet()){
-                    Input input = new Input(entry.getKey(), entry.getValue().size());
-                    list.add(input);
-                }
-                inputList = list;
-
-                inputList.sort((w1, w2) -> w2.getWordCount() - w1.getWordCount());
-
-                StringJoiner joiner = new StringJoiner("\n");
-                for (Input w : inputList) {
-                    String s = w.getValue() + " " +w.getWordCount();
-                    joiner.add(s);
-                }
-                return joiner.toString();
-            } catch (Exception e) {
-
-
-                return "Calculate Error";
-            }
+        } catch (Exception e) {
+            return "Calculate Error";
         }
     }
 
 
-    private Map<String,List<Input>> getListMap(List<Input> inputList) {
-        Map<String, List<Input>> map = new HashMap<>();
-        for (Input input :  inputList){
-//       map.computeIfAbsent(input.getValue(), k -> new ArrayList<>()).add(input);
-            if (!map.containsKey(input.getValue())){
-                ArrayList arr = new ArrayList<>();
-                arr.add(input);
-                map.put(input.getValue(), arr);
-            }
+    private String outputList(List<Input> inputCountList)
+    {
+        return inputCountList.stream()
+                .map(output -> output.getValue() + " " + output.getWordCount())
+                .collect(Collectors.joining("\n"));
+    }
 
-            else {
-                map.get(input.getValue()).add(input);
-            }
+    private List<Input> sortList(List<Input> inputList)
+    {
+        Map<String, List<Input>> map = getListMap(inputList);
+        //split the input string with 1 to n pieces of spaces
+        List<Input> list = new ArrayList<>();
+        for (Map.Entry<String, List<Input>> entry : map.entrySet()) {
+            Input input = new Input(entry.getKey(), entry.getValue().size());
+            list.add(input);
         }
+        inputList = list;
+        inputList.sort((w1, w2) -> w2.getWordCount() - w1.getWordCount());
 
-
-        return map;
+        return inputList;
     }
+    private List<Input> splitInputStringToList(String inputStr)
+    {
+        //split the input string with 1 to n pieces of spaces
 
+        return  Arrays.stream(inputStr.split("\\s+"))
+                .map(splitInputStr -> new Input(splitInputStr, 1))
+                .collect(Collectors.toList());
+    }
 
+    private Map<String,List<Input>> getListMap(List<Input> inputList) {
+        Map<String, List<Input>> map = inputList
```

Comment:
Immediately return this expression instead of assigning it to the temporary variable "map".